Scripts/Player_Scripts/player.gd

```diff
@@ -1,22 +1,33 @@
 extends CharacterBody3D
 
 
-@export var speed = 5.0
+var current_move_speed = 5.0
+@export var walking_speed = 5.0
+@export var crouch_walk_speed = 2.5
 @export var jump_velocity = 4.5
+@export var player_height = 1.7
+var b_is_crouching = false
 var mouse_sensitivity = 0.002
+var joystick_sensitivity = 2
 
 func _input(event):
 	if event is InputEventMouseMotion and Input.mouse_mode == Input.MOUSE_MODE_CAPTURED:
 		rotate_y(-event.relative.x * mouse_sensitivity)
 		$Camera3D.rotate_x(-event.relative.y * mouse_sensitivity)
 		$Camera3D.rotation.x = clampf($Camera3D.rotation.x, -deg_to_rad(70), deg_to_rad(70))
-		
+
 func _ready():
+	$RayCast3D.target_position = Vector3(0, player_height, 0)
+	$RayCast3D.enabled = true
 	Input.set_mouse_mode(Input.MOUSE_MODE_CAPTURED)
+	$CollisionShape3D.shape.height = player_height
+	print($CollisionShape3D.shape.height)
```

Comment:
print statements used for debugging should either be removed or have a high amount of verbosity.